app/lib/pdf/finance/invoice_report_data.rb

```diff
@@ -52,23 +52,22 @@ def plan
   end
 
   def logo
-    @logo_stream ||= URI.open(logo_url(:invoice))
+    @logo_file ||= logo_file(:invoice)
   rescue StandardError => exception
-    # TODO: - uncomment complete exception!
-    # rescue OpenURI::HTTPError => e
     Rails.logger.error "Failed to retrieve logo from: #{exception.message}"
     nil
   end
 
   # Depending on the storage option for the attachment, retrieve either the full path
-  # of the local file, or the URL of the file in S3
-  def logo_url(args)
+  # of the local file, or the URL of the file in S3, and read the file differently
+  def logo_file(style)
     attachment = @invoice.provider_account.profile.logo
     case attachment.options[:storage].to_sym
     when :filesystem
-      attachment.path(args)
+      # read as binary file
+      File.open(attachment.path(style), 'rb')
     when :s3
-      attachment.url(args)
+      URI.open(attachment.url(style))
```

Comment:
why change this?

Reply:
What exactly? from `open` to `URI.open` ? Rubocop was complaining about `The use of Kernel#open is a serious security risk`. I also think this is more explicit.

Reply:
Well, that's the change from one commit to another :)
And this is because I needed a different way not only to get the file locator (URL vs full path), but also a different way to read them (`URI.open` vs `File.open`), depending on whether it's a local file or an S3-hosted image.

As discussed above, I'll need to refactor this further to make sure we're closing the file descriptors.